test/test.c

```diff
@@ -62,3 +77,15 @@ UBENCH_F(c_my_fixture, strchr) {
 UBENCH_F(c_my_fixture, strrchr) {
   UBENCH_DO_NOTHING(strrchr(ubench_fixture->data, 'f'));
 }
+
+UBENCH_EX_F(c_my_fixture, strrchr_ex)
```

Comment:
I kinda think we should just not allow fixtures if you want to setup the benchmark state in the benchmark itself. One less variant, little bit cleaner!

Reply:
I would still argue that fixtures is nice to have... Let's say you have x benchmarks that you want to run on system y.
All of them would need to run create_y() and teardown_y() but the actual benchmark would want to run with different data or different functions.
I would argue for supporting this but I could go either way!

Reply:
If you think there is a use then we can keep it no bother. I just thought that it'd be this or fixtures!